mage_ai/shared/io.py

```diff
@@ -1,8 +1,24 @@
-from typing import Callable
-import aiofiles
 import os
 import shutil
 import traceback
+from typing import Callable
+
+import aiofiles
+
+
+def chmod(path, mode, append=True):
+    # recursively runs chmod against the path and either adds or sets the permission(mode)
+    try:
+        current_mode = os.stat(path).st_mode
+    except Exception as e:
+        raise e
+    if append:
+        mode = current_mode | mode
+    os.chmod(path, mode)
```

Comment:
can this operation raise an exception?

Reply:
I have wrapped it in a full try, except block and only print stack trace as with many other functions.